content/yaml-code-signing/code-signing-identities.md

```diff
@@ -0,0 +1,166 @@
+
+### Android keystores
+
+Codemagic lets you upload your Android keystores. When uploading a keystore, it is required that the keystore password, key alias, and key password (if exists) are provided.
+
+Furthermore, it is required that a unique **Reference name** is assigned to each uploaded keystore, which can be used to fetch the keystore during the build using the `codemagic.yaml` configuration file. 
+
+For each keystore, its common name, issuer, and expiration date are displayed.
+
+{{<notebox>}}
+Note that the uploaded keystore cannot be downloaded from Codemagic. It is crucial that you store a copy of the keystore file used in Codemagic as all subsequent builds released to Google Play should be signed with the same keystore.
+{{</notebox>}}
+
+## Referencing files in codemagic.yaml
+
+After uploading code signing files to Codemagic, these files can be fetched and used during the build by providing the reference names in the `codemagic.yaml` configuration.
+
+### Apple certificates and profiles
+
+Codemagic provides two means of fetching the required certificates and provisioning profiles during the build with the use of `codemagic.yaml`. Fetching can either be configured by specifying the distribution type and bundle identifier, or for more advanced use-cases, individual files can be fetched by their reference names.
+
+#### Fetching files by distribution type and bundle identifier
+
+To fetch all uploaded signing files matching a specific distribution type and bundle identifier during the build, define the `distribution_type` and `bundle_identifier` fields in your `codemagic.yaml` configuration. Note that it is necessary to configure both of the fields.
+
+Note that when using the fields `distribution_type` and `bundle_identifier`, it is not allowed to configure `provisioning_profiles` and `certificates` fields.
+
+```yaml
+environment:
+    ios_signing:
+        distribution_type: ad_hoc  # app_store | development | enterprise
+        bundle_identifier: com.example.id
+```
+
+When defining the bundle identifier `com.example.id`, Codemagic will fetch any uploaded certificates and profiles matching the extensions as well (e.g. `com.example.id.NotificationService`).
+
+#### Fetching specific files by reference names
+
+For a more advanced configuration, it is possible to pick out specific uploaded profiles and certificates for Codemagic to fetch during the build. To do so, list the references of the uploaded files under the `provisioning_profiles` and `certificates` fields, respectively. Note than when fetching individual files, the fields `distribution_type` and `bundle_identifier` are not allowed.
+
+```yaml
+environment:
+    ios_signing:
+        provisioning_profiles:
+            - profile_reference
+            - ...
+        certificates:
+            - certificate_reference
+            - ...
+```
+
+By default, Codemagic saves the files to the following locations on the build machine:
```

Comment:
> By default, ...

This hints that there is a way to save those files into some other place, which is not the case.